packages/coordinator/ts/deployer/deployer.service.ts

```diff
@@ -0,0 +1,943 @@
+          config.VkRegistry.args.messageTreeDepth,
+          config.VkRegistry.args.voteOptionTreeDepth,
+          MESSAGE_TREE_ARITY ** config.VkRegistry.args.messageBatchDepth,
+          EMode.QV,
+        ],
+      });
+
+      const isProcessNonQvVkSet = await publicClient.readContract({
+        address: vkRegistryAddress as Hex,
+        abi: VkRegistryFactory.abi,
+        functionName: "hasProcessVk",
+        args: [
+          config.VkRegistry.args.stateTreeDepth,
+          config.VkRegistry.args.messageTreeDepth,
+          config.VkRegistry.args.voteOptionTreeDepth,
+          MESSAGE_TREE_ARITY ** config.VkRegistry.args.messageBatchDepth,
+          EMode.NON_QV,
+        ],
+      });
+
+      const isTallyVkSet = await publicClient.readContract({
+        address: vkRegistryAddress as Hex,
+        abi: VkRegistryFactory.abi,
+        functionName: "hasTallyVk",
+        args: [
+          config.VkRegistry.args.stateTreeDepth,
+          config.VkRegistry.args.intStateTreeDepth,
+          config.VkRegistry.args.voteOptionTreeDepth,
+          EMode.QV,
+        ],
+      });
+
+      const isTallyNonQvVkSet = await publicClient.readContract({
+        address: vkRegistryAddress as Hex,
+        abi: VkRegistryFactory.abi,
+        functionName: "hasTallyVk",
+        args: [
+          config.VkRegistry.args.stateTreeDepth,
+          config.VkRegistry.args.intStateTreeDepth,
+          config.VkRegistry.args.voteOptionTreeDepth,
+          EMode.QV,
+        ],
+      });
+
+      if (isProcessVkSet && isProcessNonQvVkSet && isTallyVkSet && isTallyNonQvVkSet) {
+        this.logger.debug("Verifying keys are already set on the vk registry");
+      } else {
+        const gasEstimates = await kernelClient.getUserOperationGasPrice();
+        const opEstimate = await kernelClient.prepareUserOperationRequest({
+          userOperation: {
+            sender: kernelClient.account.address,
+            maxFeePerGas: gasEstimates.maxFeePerGas,
+            maxPriorityFeePerGas: gasEstimates.maxPriorityFeePerGas,
+            callData: await kernelClient.account.encodeCallData({
+              to: vkRegistryAddress as Hex,
+              value: 0n,
+              data: encodeFunctionData({
+                abi: VkRegistryFactory.abi,
+                functionName: "setVerifyingKeysBatch",
+                args: [
+                  config.VkRegistry.args.stateTreeDepth,
+                  config.VkRegistry.args.intStateTreeDepth,
+                  config.VkRegistry.args.messageTreeDepth,
+                  config.VkRegistry.args.voteOptionTreeDepth,
+                  MESSAGE_TREE_ARITY ** config.VkRegistry.args.messageBatchDepth,
+                  [EMode.QV, EMode.NON_QV],
+                  // @ts-expect-error - the abi has a more complex type for the processZkeys and tallyZkeys
+                  processZkeys,
+                  // @ts-expect-error - the abi has a more complex type for the processZkeys and tallyZkeys
```

Comment:
Will the explicit cast work here? For example: `processZkeys: processZkeys as unknown as bigint[]`

Reply:
it wants this type: readonly { alpha1: { x: bigint; y: bigint; }; beta2: { x: readonly [bigint, bigint]; y: readonly [bigint, bigint]; }; gamma2: { x: readonly [bigint, bigint]; y: readonly [bigint, bigint]; }; delta2: { x: readonly [bigint, bigint]; y: readonly [bigint, bigint]; }; ic: readonly { ...; }[]; }[]

Reply:
So, maybe it can be cast to G2Point?

Reply:
that wasn't working either